lib/src/widgets/login_body.dart

```diff
@@ -0,0 +1,256 @@
+import 'package:firebase_auth/firebase_auth.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import '../constants/constants_color.dart';
+import '../blocs/login_blocs/login_blocs.dart';
+import '../blocs/login_blocs/login_event.dart';
+import '../blocs/login_blocs/login_states.dart';
+import '../screens/home_screen.dart';
+import '../utils/validators.dart';
+import '../repositories/user_repository.dart';
+import '../constants/constants.dart';
+
+class Login_Body extends StatefulWidget {
+  final double Width;
+  final double Height;
+  Login_Body({required this.Width, required this.Height});
+  @override
+  LoginForm createState() => LoginForm(Width: Width, Height: Height);
+}
+
+class LoginForm extends State<Login_Body> {
+  // This widget is the root of your application.
+  final double Width;
+  final double Height;
+
+  LoginForm({required this.Width, required this.Height});
+  final emailController = TextEditingController();
+  final passWordController = TextEditingController();
+  final UserRepository _userRepository = UserRepository();
+  final Validators validators = Validators();
+  String checkValidateEmail(String email) {
+    bool isValidEmail = validators.isValidEmail(email);
+    if (isValidEmail == true) {
+      return "";
+    } else {
+      return "Input correct email";
+    }
+  }
+
+  String checkValidatePassword(String Password) {
+    if (Password.length < 8) {
+      return "Minimum length is 8";
+    } else if (validators.isValidPassword(Password) == false) {
+      return "Incorrect Password";
+    } else
+      return "";
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    LoginBloc bloc = BlocProvider.of<LoginBloc>(context);
+    return BlocBuilder<LoginBloc, LoginState>(
+        bloc: bloc,
+        builder: (context, state) {
+          double formPaddingTop = 0.025 * Height;
+          double formPaddingHorizontal = 0.0667 * Width;
+          double mailRemindFontSize = 14;
+          double emailInputPadding = Height * 0.046;
+          double emailInputHorizontalPadding = 0.0667 * Width;
+          double emailFontSize = 14;
+          double emailRemindContainerWidth = Width * 0.53;
+          double emailRemindContainerHeight = 22;
+          double passwordVerticalPadding = Height * 0.037;
+          double passwordHorizontalPadding = Width * 0.067;
+          double passwordFontSize = 14;
+          double loginHeightContainer = 70;
+          double loginButtonHorizontalPadding = 0.05 * Width;
+          double loginFontSize = 16;
+          double newScratchPadding = 0.037 * Height;
```

Comment:
why we have these fomulars?

Reply:
It is the ratio of each container , it is for multiple devices have multiple size thus we choose the fraction of each container for suitable for each mobiles or tablets